src/generators/observe.js

```diff
@@ -6,6 +6,10 @@ export default function(initialize) {
   let resolve;
   const dispose = initialize(change);
 
+  if (dispose != null && typeof dispose !== "function") {
+    throw new Error("invalid dispose");
```

Comment:
Something like this?

```suggestion
    throw new Error(typeof dispose.then === "function"
        ? "async initializers are not supported"
        : "initializer returned something, but not a dispose function");
```

Reply:
Yeah, this is greatly helpful!